### PR TITLE
Add back-matter 'has' constraints

### PR DESCRIPTION
**Pull request by Gabeblis**
# Committer Notes

### Description 
Added back-matter 'has' constraints and tests for each of the constraints. These should be added as they are part of the effort write all of the constraints from the constraint tracker.

### Changes 
- Added constraints has-user-guide, has-rules-of-behavior, has-information-system-contingency-plan, has-configuration-management-plan, has-incident-response-plan, has-separation-of-duties-matrix to fedramp-external-constraints.xml.  
- Added pass and fail yaml tests for all of the above constraints.  
- Edited ssp-all-VALID to ensure all constraints pass correctly.  

{Please provide a description of what this PR accomplishes. Be sure to reference any issues addressed. If the PR is a work-in-progress submitted for early review, please submit the PR as a draft PR using the "Draft pull request" dropdown.}

### All Submissions:

- [x] Have you selected the correct base branch per [Contributing](https://github.com/GSA/fedramp-automation/blob/master/CONTRIBUTING.md) guidance?
- [x] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/GSA/fedramp-automation/pulls) for the same update/change?
- [ ] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] If applicable, have all [FedRAMP Documents Related to OSCAL Adoption](https://github.com/GSA/fedramp-automation) affected by the changes in this issue have been updated.?
- [ ] If applicable, does this PR reference the issue it addresses and explain how it addresses the issue?

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### CODE REVIEW

**Review by Rene2mt — DISMISSED** [review]

File: src/validations/constraints/fedramp-external-constraints.xml [hunk 1]
```diff
@@ -30,7 +30,24 @@
         <expect id="resource-has-base64-or-rlink" target="back-matter/resource" test="count(./rlink) >= 1 or count(./base64) >= 1" level="WARNING">
             <message>Every supporting artifact found in a citation must have at least one base64 or rlink element.</message>
         </expect>
+        <expect id="has-user-guide" target="back-matter" test="resource[prop[@name eq 'type' and @value eq 'users-guide']]" level="WARNING">
```
Comment (aj-stein-gsa) [inline src/validations/constraints/fedramp-external-constraints.xml:33]: This section of attachment constraints are great. We may want to make sure we align these attachment checks with the current public list of required documents.

https://automate.fedramp.gov/documentation/ssp/5-attachments/

Comment (aj-stein-gsa) [inline src/validations/constraints/fedramp-external-constraints.xml:33]: And also, I am not sure we want _only_ a warning and not an error? These are important requirements for FR review, did the previous Schematron rules mark them warning too? (I know a reason that could have been when I was last here, but not the latest of the Schematron rules).

**Comment by Gabeblis** [comment]
> This is all good, do we have docs on automate.fedramp.gov to align with the attachment requirements for a SSP? In [the SSP guidance](https://automate.fedramp.gov/documentation/ssp/4-ssp-template-to-oscal-mapping), we call out an interconnection agreement, network diagram, and dataflow diagram, but not many of the attachments needed here.

All of them except for has-separation-of-duties-matrix are addressed here: https://automate.fedramp.gov/documentation/ssp/5-attachments/ 
There is an issue (https://github.com/GSA/fedramp-automation/issues/534) and draft PR (https://github.com/GSA/fedramp-automation/pull/594) for adding documentation for separation of duties.

**Review by aj-stein-gsa — CHANGES_REQUESTED** [review]

File: src/validations/constraints/fedramp-external-constraints.xml (hunk 1 above)
Comment (Gabeblis) [reply src/validations/constraints/fedramp-external-constraints.xml:33]: These are all currently marked as a warning in the Schematron.

Comment (Gabeblis) [reply src/validations/constraints/fedramp-external-constraints.xml:33]: > This section of attachment constraints are great. We may want to make sure we align these attachment checks with the current public list of required documents.
> 
> https://automate.fedramp.gov/documentation/ssp/5-attachments/

These constraints align with those checks. Should we mention that they are required in the SSP, or is that implied?

Comment (aj-stein-gsa) [reply src/validations/constraints/fedramp-external-constraints.xml:33]: Ah I see your above comment, disregard!

**Comment by aj-stein-gsa** [comment]
> > This is all good, do we have docs on automate.fedramp.gov to align with the attachment requirements for a SSP? In [the SSP guidance](https://automate.fedramp.gov/documentation/ssp/4-ssp-template-to-oscal-mapping), we call out an interconnection agreement, network diagram, and dataflow diagram, but not many of the attachments needed here.
> 
> All of them except for has-separation-of-duties-matrix are addressed here: https://automate.fedramp.gov/documentation/ssp/5-attachments/ There is an issue (#534) and draft PR (#594) for adding documentation for separation of duties.

OK then we are in a good place, I will re-review in a bit.

File: src/validations/constraints/fedramp-external-constraints.xml (hunk 1 above)
Comment (aj-stein-gsa) [reply src/validations/constraints/fedramp-external-constraints.xml:33]: I had some discussions with @david-waltermire and thought about it some given our docs. You cannot really submit a valid SSP without them, so we need these to be ERROR.

Comment (aj-stein-gsa) [inline src/validations/constraints/fedramp-external-constraints.xml:33]: See https://github.com/GSA/fedramp-automation/pull/654/files#r1754412173.

```suggestion
        <expect id="has-user-guide" target="back-matter" test="resource[prop[@name eq 'type' and @value eq 'users-guide']]" level="ERROR">
```

File: src/validations/constraints/fedramp-external-constraints.xml [hunk 2]
```diff
@@ -30,7 +30,24 @@
...
+            <message>A FedRAMP SSP must have a User Guide attached.</message>
+        </expect>
+        <expect id="has-rules-of-behavior" target="back-matter" test="resource[prop[@name eq 'type' and @value eq 'rules-of-behavior']]" level="WARNING">
```
Comment (aj-stein-gsa) [inline src/validations/constraints/fedramp-external-constraints.xml:36]: See https://github.com/GSA/fedramp-automation/pull/654/files#r1754412173.

```suggestion
        <expect id="has-rules-of-behavior" target="back-matter" test="resource[prop[@name eq 'type' and @value eq 'rules-of-behavior']]" level="ERROR">
```

File: src/validations/constraints/fedramp-external-constraints.xml [hunk 3]
```diff
@@ -30,7 +30,24 @@
...
+            <message>A FedRAMP SSP must have Rules of Behavior.</message>
+        </expect>
+        <expect id="has-information-system-contingency-plan" target="back-matter" test="resource[prop[@name eq 'type' and @value eq 'plan' and @class eq 'information-system-contingency-plan']]" level="WARNING">
```
Comment (aj-stein-gsa) [inline src/validations/constraints/fedramp-external-constraints.xml:39]: See https://github.com/GSA/fedramp-automation/pull/654/files#r1754412173.

```suggestion
        <expect id="has-information-system-contingency-plan" target="back-matter" test="resource[prop[@name eq 'type' and @value eq 'plan' and @class eq 'information-system-contingency-plan']]" level="ERROR">
```

File: src/validations/constraints/fedramp-external-constraints.xml [hunk 4]
```diff
@@ -30,7 +30,24 @@
...
+            <message>A FedRAMP SSP must have a Contingency Plan attached.</message>
+        </expect>
+        <expect id="has-configuration-management-plan" target="back-matter" test="resource[prop[@name eq 'type' and @value eq 'plan' and @class eq 'configuration-management-plan']]" level="WARNING">
```
Comment (aj-stein-gsa) [inline src/validations/constraints/fedramp-external-constraints.xml:42]: See https://github.com/GSA/fedramp-automation/pull/654/files#r1754412173.

```suggestion
        <expect id="has-configuration-management-plan" target="back-matter" test="resource[prop[@name eq 'type' and @value eq 'plan' and @class eq 'configuration-management-plan']]" level="ERROR">
```

File: src/validations/constraints/fedramp-external-constraints.xml [hunk 5]
```diff
@@ -30,7 +30,24 @@
...
+            <message>A FedRAMP SSP must have a Configuration Management Plan attached.</message>
+        </expect>
+        <expect id="has-incident-response-plan" target="back-matter" test="resource[prop[@name eq 'type' and @value eq 'plan' and @class eq 'incident-response-plan']]" level="WARNING">
```
Comment (aj-stein-gsa) [inline src/validations/constraints/fedramp-external-constraints.xml:45]: See https://github.com/GSA/fedramp-automation/pull/654/files#r1754412173.

```suggestion
        <expect id="has-incident-response-plan" target="back-matter" test="resource[prop[@name eq 'type' and @value eq 'plan' and @class eq 'incident-response-plan']]" level="ERROR">
```

File: src/validations/constraints/fedramp-external-constraints.xml [hunk 6]
```diff
@@ -30,7 +30,24 @@
...
+            <message>A FedRAMP SSP must have an Incident Response Plan attached.</message>
+        </expect>
+        <expect id="has-separation-of-duties-matrix" target="back-matter" test="resource[prop[@name eq 'type' and @value eq 'separation-of-duties-matrix']]" level="WARNING">
```
Comment (aj-stein-gsa) [inline src/validations/constraints/fedramp-external-constraints.xml:48]: See https://github.com/GSA/fedramp-automation/pull/654/files#r1754412173.

```suggestion
        <expect id="has-separation-of-duties-matrix" target="back-matter" test="resource[prop[@name eq 'type' and @value eq 'separation-of-duties-matrix']]" level="ERROR">
```

**Review by aj-stein-gsa — CHANGES_REQUESTED** [review]
See https://github.com/GSA/fedramp-automation/pull/654/files#r1754412173 for updated review and proposed changes.

File: src/validations/constraints/fedramp-external-constraints.xml (hunk 1 above)
Comment (Gabeblis) [reply src/validations/constraints/fedramp-external-constraints.xml]: > See https://github.com/GSA/fedramp-automation/pull/654/files#r1754412173.

Adressed in 393b1dbdc1edd1fb49bbe9e244f10f4f04454973

File: src/validations/constraints/fedramp-external-constraints.xml (hunk 2 above)
Comment (Gabeblis) [reply src/validations/constraints/fedramp-external-constraints.xml]: adressed in 393b1dbdc1edd1fb49bbe9e244f10f4f04454973

File: src/validations/constraints/fedramp-external-constraints.xml (hunk 6 above)
Comment (Gabeblis) [reply src/validations/constraints/fedramp-external-constraints.xml]: > See https://github.com/GSA/fedramp-automation/pull/654/files#r1754412173.

adressed in 393b1dbdc1edd1fb49bbe9e244f10f4f04454973

File: src/validations/constraints/fedramp-external-constraints.xml (hunk 5 above)
Comment (Gabeblis) [reply src/validations/constraints/fedramp-external-constraints.xml]: > See https://github.com/GSA/fedramp-automation/pull/654/files#r1754412173.

addressed in 393b1dbdc1edd1fb49bbe9e244f10f4f04454973

File: src/validations/constraints/fedramp-external-constraints.xml (hunk 4 above)
Comment (Gabeblis) [reply src/validations/constraints/fedramp-external-constraints.xml]: > See https://github.com/GSA/fedramp-automation/pull/654/files#r1754412173.

addressed in 393b1dbdc1edd1fb49bbe9e244f10f4f04454973

File: src/validations/constraints/fedramp-external-constraints.xml (hunk 3 above)
Comment (Gabeblis) [reply src/validations/constraints/fedramp-external-constraints.xml]: > See https://github.com/GSA/fedramp-automation/pull/654/files#r1754412173.

addressed in 393b1dbdc1edd1fb49bbe9e244f10f4f04454973

**Review by aj-stein-gsa — APPROVED** [review]

**Review by aj-stein-gsa — APPROVED** [review]

**Pull request merged**